packages/cmf/src/settings.js

```diff
@@ -17,28 +18,39 @@ export function attachRef(state, obj) {
 	if (props._ref) {
 		const ref = state.cmf.settings.ref[props._ref];
 		invariant(ref, `CMF/Settings: Reference '${props._ref}' not found`);
-		props = Object.assign(
-			{},
-			state.cmf.settings.ref[props._ref],
-			obj
-		);
+		props = Object.assign({}, state.cmf.settings.ref[props._ref], obj);
 		delete props._ref;
 	}
 	return props;
 }
 
 export function attachRefs(state, props) {
 	const attachedProps = attachRef(state, props);
-	Object.keys(attachedProps).forEach(
-		(key) => {
-			attachedProps[key] = attachRef(state, attachedProps[key]);
-		}
-	);
+	Object.keys(attachedProps).forEach(key => {
+		attachedProps[key] = attachRef(state, attachedProps[key]);
+	});
 	return attachedProps;
 }
 
+export function nonMemoizedMapStateToViewProps(state, ownProps, componentName, componentId) {
+	let viewProps = {};
+	let viewId = ownProps.view;
+	if (!ownProps.view && componentName && !componentId) {
+		viewId = componentName;
+	} else if (!ownProps.view && componentName && componentId) {
+		viewId = `${componentName}:${componentId}`;
+	}
+	if (viewId && state.cmf.settings.views[viewId]) {
```

Comment:
this seems fishy as well, could you pls explain what this check should do?
maybe it can be simplified to (state.cmf.settings.views[viewId]) ?

Reply:
it is done this way because of
```
const data = {
  "undefined": "value"
};

console.log(data[undefined]);
```